mobile_packet_verifier/src/burner.rs

```diff
@@ -0,0 +1,266 @@
+        {
+            // Fetch the sub dao epoch info:
+            let epoch = self.provider.get_epoch_info().await?.epoch;
+            let (sub_dao_epoch_info, _) = Pubkey::find_program_address(
+                &[
+                    "sub_dao_epoch_info".as_bytes(),
+                    self.program_cache.sub_dao.as_ref(),
+                    &epoch.to_le_bytes(),
+                ],
+                &helium_sub_daos::ID,
+            );
+
+            let amount = bytes_to_dc(total_bytes);
+
+            // Burn the DC for the payer
+            let ddc_key = crate::pdas::delegated_data_credits(&self.program_cache.sub_dao, &payer);
+            let account_data = self.provider.get_account_data(&ddc_key).await?;
+            let mut account_data = account_data.as_ref();
+            let escrow_account =
+                DelegatedDataCreditsV0::try_deserialize(&mut account_data)?.escrow_account;
+
+            tracing::info!("Burning {} DC from {}", amount, payer);
+
+            let instructions = {
+                let request = RequestBuilder::from(
+                    data_credits::id(),
+                    &self.cluster,
+                    std::rc::Rc::new(Keypair::from_bytes(&self.keypair).unwrap()),
+                    Some(CommitmentConfig::confirmed()),
+                    RequestNamespace::Global,
+                );
+
+                let accounts = accounts::BurnDelegatedDataCreditsV0 {
+                    sub_dao_epoch_info,
+                    dao: self.program_cache.dao,
+                    sub_dao: self.program_cache.sub_dao,
+                    account_payer: self.program_cache.account_payer,
+                    data_credits: self.program_cache.data_credits,
+                    delegated_data_credits: crate::pdas::delegated_data_credits(
+                        &self.program_cache.sub_dao,
+                        &payer,
+                    ),
+                    token_program: spl_token::id(),
+                    helium_sub_daos_program: helium_sub_daos::id(),
+                    system_program: solana_program::system_program::id(),
+                    dc_burn_authority: self.program_cache.dc_burn_authority,
+                    dc_mint: self.program_cache.dc_mint,
+                    escrow_account,
+                    registrar: self.program_cache.registrar,
+                };
+                let args = instruction::BurnDelegatedDataCreditsV0 {
+                    args: data_credits::BurnDelegatedDataCreditsArgsV0 { amount },
+                };
+
+                // As far as I can tell, the instructions function does not actually have any
+                // error paths.
+                request
+                    .accounts(accounts)
+                    .args(args)
+                    .instructions()
+                    .unwrap()
+            };
+
+            let blockhash = self.provider.get_latest_blockhash().await?;
+            let signer = Keypair::from_bytes(&self.keypair).unwrap();
```

Comment:
why do we convert this keypair to bytes when we get a new `Burner` if we're converting it back to bytes in the two places where we actually use it? couldn't we just stash the `Keypair` and avoid the potentially fallible `unwrap`s?

Reply:
Keypair does not implement Copy or Clone unfortunately.